homeassistant/components/renson_healthbox3/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for Renson integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from pyhealthbox3.healthbox3 import (
+    Healthbox3,
+    Healthbox3ApiClientAuthenticationError,
+    Healthbox3ApiClientCommunicationError,
+    Healthbox3ApiClientError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_create_clientsession
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(CONF_API_KEY): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Renson."""
+
+    VERSION = 1
+
+    async def validate_input(
+        self, hass: HomeAssistant, data: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Validate the user input allows us to connect."""
+
+        if CONF_API_KEY in data:
+            await self._test_credentials(
+                ipaddress=data[CONF_HOST],
+                apikey=data[CONF_API_KEY],
+            )
+        else:
+            await self._test_connectivity(ipaddress=data[CONF_HOST])
+
+        return {"title": "Renson"}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await self.validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except Healthbox3ApiClientAuthenticationError:
+            errors["base"] = "auth"
+        except Healthbox3ApiClientCommunicationError:
+            errors["base"] = "connection"
+        except Healthbox3ApiClientError:
+            errors["base"] = "unknown"
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title=info["title"], data=user_input)
```

Comment:
The title seems to be fixed.

```suggestion
            return self.async_create_entry(title="Renson", data=user_input)
```